server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -192,7 +191,8 @@ private void uploadBlob(
                 listener.onFailure(new FileTransferException(fileSnapshot, ex));
             });
 
-            Objects.requireNonNull(fileSnapshot.getChecksum());
+            // Only the first generation doesn't have checksum
+            assert (fileSnapshot.getChecksum() != null || fileSnapshot.getName().contains("-1."));
```

Comment:
is it possible that first generation will not have `1` as generation number? I am thinking from segments points of view. Whenever we create shard for the first time, the `segments_N` that get created is `segments_4` (or sometimes `segments_3`). Can the same happen to translog? Is there a way to make this change agnostic to the filename? If file is empty, can we safely ignore the checksum check?

Reply:
>Can the same happen to translog?

It should not as of now . The first generation always has `1` as generation number . 

Refer 

https://github.com/opensearch-project/OpenSearch/blob/f1825fd45bd56e368391994f1310e1d18231e1b3/server/src/main/java/org/opensearch/index/translog/Translog.java#L2038

>If file is empty, can we safely ignore the checksum check?

The file is not empty , it always has headers consisting of uuid . For checking number of operations , we would need to read the whole file.